client/shaders/second_stage/opengl_fragment.glsl

```diff
@@ -58,6 +59,14 @@ vec4 applyToneMapping(vec4 color)
 	color.rgb *= whiteScale;
 	return color;
 }
+
+vec3 applySaturation(vec3 color, float factor)
+{
+	// Calculate the perceived luminosity from the RGB color.
+	// See also: https://www.w3.org/WAI/GL/wiki/Relative_luminance
+	float brightness = dot(color, vec3(0.2125, 0.7154, 0.0721));
+       	return mix(vec3(brightness), color, factor);
```

Comment:
Tabs vs. spaces.

Reply:
@lhofhansl Line 68 still uses spaces for indentation, unlike the rest of the file.

Reply:
Fixed now